docs/Tutorials and concepts/Concepts/Out-parameter and its purpose.md

```diff
@@ -0,0 +1,59 @@
+In ImageJS several functions have an optional parameter called `out`.
+This parameter allows to choose which image to use as an output. It enables better memory management by allowing writing image into an existing one instead of cloning it every time a function is executed.
+
+This option is mostly supported for filters which take a single image as input and produce a single image as output.
+
+For overwriting to be successful,`out` must be compatible with target width, height, depth and color model.
+
+Here is a simple example of correct and incorrect usage of `out` while applying `invert` filter:
+
+```ts
+//Creates a gray image of width = 3 and height = 3.
+let image1 = testUtils.createGreyImage([
+  [1, 1, 1],
+  [1, 1, 1],
+  [1, 1, 1],
+]);
+//Creates an RGB image of width = 1 and height = 3.
+let image2 = testUtils.createRgbImage([
+  [1, 1, 1],
+  [1, 1, 1],
+  [1, 1, 1],
+]);
+//Image to apply filter on.
+let testImage = testUtils.createGreyImage([
+  [1, 1, 1],
+  [2, 2, 2],
+  [3, 3, 3],
+]);
```

Comment:
It's not great to write the examples with `testUtils`. It's not something which is included in the published library. It's also quiet verbose.

Can you create this example by simply using `new Image(width, height, options)`. The values of the pixels do not matter to make your point.